src/internal/orders/sales-order-notification.ts

```diff
@@ -0,0 +1,32 @@
+import { hideAndFreeze, Joi, _internal, DateTimeZone } from "../common";
+import { SalesOrderIdentifier, SalesOrderIdentifierBase } from "./sales-order-identifier";
+
+export interface SalesOrderNotificationPOJO extends SalesOrderIdentifier {
+  orderNumber: string;
+
+  importedDate: DateTimeZone | Date | string;
+}
+
+export class SalesOrderNotification extends SalesOrderIdentifierBase{
+  // TODO: Add better validation and unit tests
+  public static readonly [_internal] = {
+    label: "sales order notification",
+    schema: SalesOrderIdentifier[_internal].schema.keys({
+      orderNumber: Joi.string().required(),
```

Comment:
How is `orderNumber` different from `id` (on `SalesOrderIdentifierBase`).  I think they're the same thing, so I think `orderNumber` isn't needed